### PR TITLE
Recognise degenerate GUI colors

### PR DESCRIPTION
**Pull request by jpirnay**
if for whatever reasons GUI colors become degenerate, i.e. identical to the bedcolor, then the colorschme will be reset to default

### CODE REVIEW

File: meerk40t/gui/scene/guicolors.py [hunk 1]
```diff
@@ -35,6 +35,26 @@ def __init__(self, context):
...
+        self.sanity_check()
+
+    def sanity_check(self):
+        # Look at some color-combinations to see if there are identical colors, if that's the case
+        # then this is degenerate and will lead to default colors
+        def identical(color1, color2):
+            return color1.GetRGB() == color2.GetRGB()
+        degenerate = False
+        degenerate = degenerate or identical(self.color_bed, self.color_guide)
+        degenerate = degenerate or identical(self.color_bed, self.color_grid)
```
[the remark this answers is not in the document]
Comment (tatarize) [reply meerk40t/gui/scene/guicolors.py:47]: If I set a grid line white it should destroy everything.

**Review by tatarize — CHANGES_REQUESTED** [review]

**Comment by tatarize** [comment]
Paiin's use case is that his old Raspberry pi hasn't been updated in ages, and he used it during those few cases where it would have been weird. This seems like it flags valid color settings and undoes the entire theme.

**Review by tatarize — APPROVED** [review]
It's certainly not terrible. I don't love the inability to set bed and grid-lines white. Also, apparently you can set colors degenerate if you specify a color that doesn't exist. Which is weird I would expect it to not do anything in that case.

**Comment by jpirnay** [comment]
Okay I'll rework it just to reset to invalidate the degenerate colors not the full theme

**Review by tatarize — APPROVED** [review]

**Pull request merged**